### PR TITLE
Restrict EnableGenerateDocumentationFile to .NET 8 warning wave

### LINKED ISSUE #70460: EnableGenerateDocumentationFile analyzer is a breaking change
The diagnostic added in #58835 is an unconditional diagnostic that fires on existing code. That means it represents a breaking change to customers when upgrading to a new version of the .NET SDK toolset. This was discovered by 1P customers trying to upgrade and represents a blocker for them. This diagnostic needs to be conditioned on a warning version at the least.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #70460

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -114,6 +114,9 @@ private void AnalyzeCompilation(CompilationAnalysisContext context)
             // in their project file to enable IDE0005 on build.
 
             var compilation = context.Compilation;
+            if (compilation.Options.WarningLevel < 8)
+                return;
```
Comment (sharwell) [inline src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs:118]: 📝 Note that this warning only gets reported if the user has `EnforceCodeStyleInBuild` enabled. For this scenario, the analyzer will come from the _SDK_ and not from Visual Studio itself, so this fix will only roll out to users once they have a .NET SDK version which contains the fix.

**Review by mavasani — APPROVED** [review]

**Review by jaredpar — APPROVED** [review]

File: src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs (hunk 1 above)
Comment (333fred) [reply src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs:118]: > It's also the first time we've tied a non-compiler warning to a compiler warning wave level.

I do wonder about that: is there an easy way we can tie it to `AnalysisLevel` instead?

Comment (sharwell) [reply src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs:118]: What is AnalysisLevel?

Comment (333fred) [reply src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs:118]: `AnalysisLevel` is the more general MSBuild flag that influences all built-in analyzers, and also sets `WarningLevel`. It seems to me to be the more accurate thing to use here.

**Review by arkalyanms — APPROVED** [review]

File: src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs (hunk 1 above)
Comment (arkalyanms) [reply src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs:118]: is this a blocker for landing this PR? 

(we do need a second compiler signoff)

Comment (333fred) [reply src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs:118]: Sam did some investigating, and it seems like we can't easily do this change, so I'm ok with proceeding here.

**Review by 333fred — APPROVED** [review]

**Comment by arkalyanms** [comment]
@sharwell planning to merge this to catch a 17.9 preview train before GA. I know you were planning on adding an automation here, but we can target that against 17.9 and backport in the future as necessary.

**Pull request merged**